plugin/trace/README.md

```diff
@@ -38,6 +39,8 @@ trace [ENDPOINT-TYPE] [ENDPOINT] {
 * `service` **NAME** allows you to specify the service name reported to the tracing server.
   Default is `coredns`.
 * `client_server` will enable the `ClientServerSameSpan` OpenTracing feature.
+* `tag` is used to emit custom tags with each trace, it can be included multiple times
+  If the `metadata` plugin used the value can be dynamic, use the format `{prefix/name}`
```

Comment:
the format should prolly be `{trace/tagname}`?

Reply:
> Where do you implement the metadata.Provider interface ?

I don't. I'm not adding any new metadata from the `trace` plugin, I'm allowing it to be read from other plugins by using the syntax:

```
tag tagName {someOtherPlugin/dynamicValue}
```

The `prefix` should be the name of the plugin that *writes* the value (not `trace` which only *reads* the value).

I may have misunderstood what @johnbelamaric was asking for, but I believe what he wanted was the ability to read tag values dynamically from other plugins via the metadata API.

The implementation is based on the `rewrite` plugin which also doesn't implement `metadata.Provider`, but reads values from it if the `{...}` syntax is used in the rewrite rules.